inst/scripts/insert_image.R

```diff
@@ -0,0 +1,83 @@
+local({
+  txt_input = function(..., width = '100%') shiny::textInput(..., width = width)
+
+  blogfile = rstudioapi::getSourceEditorContext()
+  if (blogfile$path == '') stop('Please select the blog post source file before using this addin', call. = FALSE)
+
+  blogpath = normalizePath(blogfile$path)
+  imgdir = file.path(
+    gsub('/content/.*', '', blogpath),
+    'static',
+    dirname(gsub('.*content/', '', blogpath)),
+    paste0(xfun::sans_ext(basename(blogpath)), '_files')
+  )
+
+  shiny::runGadget(
+    miniUI::miniPage(miniUI::miniContentPanel(
+      shiny::fillRow(
+        shiny::fileInput('newimg', 'Image', placeholder = 'Select external image'),
+        height = '90px'
+      ),
+      shiny::fillRow(
+        txt_input('imgwidth', 'Image width', '', '(optional) Example: 400 for 400px'),
+        txt_input('imgheight', 'Image height', '', '(optional) Example: 200 for 200px'),
+        height = '70px'),
+      shiny::fillRow(
+        txt_input('imgalt', 'Image alternative text', '', '(optional) Example: awesome screenshot'),
+        height = '70px'
+      ),
+      shiny::fillRow(
+        txt_input('imgdir', 'Target image location (customize only if necessary)', imgdir),
```

Comment:
I meant the path to the image instead of the parent dir of the image.

After an image is uploaded, there should be a `shiny::updateTextInput()` to update the path in this text input.

Reply:
Oh one more thing: I hope to strip the full path before `static/`, i.e. only show `static/post/..../foo.png` in the text field. To the author, the full path is probably not very useful.

Reply:
I now only keep the path starting from `static` and several of the changes as I interpreted them. Thx for the tips/suggestions!